application/src/main/java/net/corda/v5/application/interop/parameters/ParameterType.java

```diff
@@ -2,19 +2,38 @@
 
 import org.jetbrains.annotations.NotNull;
 
+/**
+ * A [ParameterType] is the type of [TypedParameter]. It is always one of a small set of primitive types, or
+ * a [QualifiedType] qualifying a primitive type with a [TypeQualifier] which identifies a more complex type.
+ */
 public interface ParameterType<T> {
     @NotNull
     ParameterTypeLabel getTypeLabel();
 
+    /**
+     * @return The expected type of [TypedParameterValue] for this [ParameterType]
+     */
     @NotNull
     Class<T> getExpectedType();
 
+    /**
+     * @return The boolean which determines whether the ParameterType is primitive or complex.
```

Comment:
You actually didn't explain which boolean values relates to which "state". Also you are using word `complex`, is this the same as `qualified`?